### DISCUSSION TITLE
gitlens.remotes, how to add multiple gerrit domains

### DISCUSSION THREAD

**Discussion by Helban** [gitkraken/vscode-gitlens#2803] [Q&A]
Hi!

I want to know, how can I add multiple gerrit sources, something like this: (doesnt seem to work)
I need that since some of the oldest changes are moved to the archive so to speak with a separate url.

![image](https://github.com/gitkraken/vscode-gitlens/assets/6530012/44307155-e87c-4bae-b37b-db9e16b40733)

And then have 2 icons visible under the hover pop up and under the "..." 

![image](https://github.com/gitkraken/vscode-gitlens/assets/6530012/1d27f1d8-b600-4276-8567-3407caf1b256)

And here, duplicate the part about recent Gerrit, but do it for the archive.:

![image](https://github.com/gitkraken/vscode-gitlens/assets/6530012/cfb103c3-2706-4e0e-ba92-b32b04d792c7)

Is that even possible?

**Comment by Helban** [comment]
Would really like to know if what I want is even possible, perhaps this should be a feature request?

**Comment by eamodio** [comment]
GitLens doesn't show multiple icons, but if there are multiple matches and no default remote is set, then the option will result in a quick pick menu to choose the desired remote.

**Reply by Helban** [reply]
Hmm... it seems like the second entry above doesn't work as a "gerrit" type? I mean, it's gerrit, but the address has "-archive" added to it, and when i just leave the second entry in the "gerrit.remotes" it doesnt work, the icon doesnt show. 
Does the "gerrit" type only accepts specific addresses? Can i change that somehow, while maintaining the gerrit type functionality of the rest?

**Reply by eamodio** [reply]
@kramarz89 can you post the output of `git remote -v` for that repo? if the URL must be private, please obfuscate it, but keep as much of the structure as possible so I can test the parsing on something relatively identical to what you have.

**Reply by Helban** [reply]
└─ $ ▶ git remote -v
origin  ssh://user@gerritmirror-ha.address.at.company.com:29418/repository/abc (fetch)
origin  ssh://user@gerrit.company.com:29418/repository/abc (push)


only the second "push" address, that is `gerrit.company.com` works when i put it into the `gitlens.remotes`.
But the thing is, this address has only commits that are within a year. Anything older than a year is only visible at `gerrit-archive.thing.anotherthing.company.com` (dont ask my why my company made it that way) which is not the same address as the above "fetch" part or the "push" part

**Reply by eamodio** [reply]
@kramarz89 Thanks for those details -- GitLens can only support 1 remote provider integration per remote name. But in tomorrow's pre-release version, I've improved our parsing to better handle the case where the first URL we see doesn't match a remote provider, but the 2nd does. Let me know if that improves things for you.

**Reply by Helban** [reply]
I switched to pre-release however not sure if the change is in already. Anyway, It's not the problem that with 2 entries in the `gitlens.remotes` it's not working, the problem is that it's not working for the archive version of gerrit, where the address is different than anything thats found under `git remote -v`

```
"gitlens.remotes": [
      {
        "domain": "gerrit-archive.thing.anotherthing.company.com",
        "type": "Gerrit",
        "name": "Archive Gerrit",
        "protocol": "https"
      }
```

this alone doesnt work, and this domain, is not found under `git remote -v` its completely different.

```
git remote -v
origin ssh://user@gerritmirror-ha.address.at.company.com:29418/repository/abc (fetch)
origin ssh://user@gerrit.company.com:29418/repository/abc (push)
```

did a small test, and if i put the (fetch)`gerritmirror-ha.address.at.company.com` domain into the settings, it works, probably because it can be found in the `git remote -v`  but that domain is not the gerrit archive. So in other words, I would like it to work for any domain address, even if it's not found under `git remote -v` and still work and behave like a gerrit domain. the gerrit-archive domain is read-only, so it's not possible for me to add it to `git remote -v` would make no sense whatsoever.

Also, when i had the popup to choose which one i want to open, i picked the gerritmirror one, and now it saved "as the default", if i remove it, then it opens the `gerrit.company.com` but when i add it again, it comes back as default, no pop up so doesnt ask me again to pick, and i want to always have the option to open either one of them, since like i've mentioned, the archive is for old commits.

**Reply by eamodio** [reply]
GitLens requires that the custom remotes match a configured remote. So, I think the only way to get this to work the way you'd want, is to fully define a remote of type "Custom" and provide all the `urls` to map it as you need. The `domain` would have to match a remote, but then your URLs could point to a different place.

**Reply by Helban** [reply]
Ok I understand. That workaround works. But now i would like to be able to always have the pop up option to either choose one or the other entry. I had the pop up once, the first time i've choosen one entry, now every time that entry is in the `gitlens.remotes` then it's default and the only option. How can I change that?
